llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -694,11 +697,8 @@ class VPLiveOut : public VPUser {
     return U->getVPUserID() == VPUser::VPUserID::LiveOut;
   }
 
-  /// Fixup the wrapped LCSSA phi node in the unique exit block.  This simply
-  /// means we need to add the appropriate incoming value from the middle
-  /// block as exiting edges from the scalar epilogue loop (if present) are
-  /// already in place, and we exit the vector loop exclusively to the middle
-  /// block.
+  /// Fixup the wrapped LCSSA phi node. This means we need to add the
+  /// appropriate incoming value from the precessor Pred.
```

Comment:
```suggestion
  /// Fix the wrapped phi node. This means adding an incoming value to exit block phi's from the vector loop via middle block (values from scalar loop already reach these phi's), and updating the value to scalar header phi's from the scalar preheader.
```
Note that a phi (`ResumePhi`?) could be added inside middle block for the former case, thereby having VPlan maintain LCSSA form, and ensuring parent of incoming value is always the desired predecessor. Can be done as a follow-up (or preparatory?) patch if preferred.